src/setfit/onnx.py

```diff
@@ -0,0 +1,227 @@
+from .modeling import SetFitModel
+from sentence_transformers import models
+import numpy as np
+import onnx
+import warnings
+from typing import Optional, Union
+import torch
+from sklearn.linear_model import LogisticRegression
+
+
+class ONNXSetFitModel(torch.nn.Module):
+    def __init__(
+        self,
+        model_body,
+        pooler,
```

Comment:
What do you think about defining a function like this:

```python
def mean_pooling(token_embeddings, attention_mask) -> torch.Tensor:
    input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
    return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(input_mask_expanded.sum(1), min=1e-9)
```

and using it as the default for the `pooler`? I took this from the model card of `sentence-transformers`: https://huggingface.co/sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2#usage-huggingface-transformers